File: src/_03_BehavioralPattern/_03_01_ChainOfResponsibilities/after/LoggingRequestHandler.java
package _03_BehavioralPattern._03_01_ChainOfResponsibilities.after;

public class LoggingRequestHandler extends RequestHandler {

  public LoggingRequestHandler(RequestHandler nextHandler) {
    super(nextHandler);
  }

  @Override
  public void handler(Request request) {
    System.out.println("로깅 작업 완료");
    super.handler(request);
  }
  
}

File: src/_03_BehavioralPattern/_03_01_ChainOfResponsibilities/LoggingRequestHandler.js
import RequestHandler from "./RequestHandler.js";

/**
 * @typedef { import("./Request").default } Request
 */

export default class LoggingRequestHandler extends RequestHandler {
  /** @param { RequestHandler } nextHandler */
  constructor(nextHandler) {
    super(nextHandler);
  }

  /**
   * @override
   * @param { Request } request 
   */
  handler(request) {
    console.log("로깅 되었습니다.");
    super.handler(request);
  }
}